pkgs/development/libraries/libbacktrace/default.nix

```diff
@@ -1,22 +1,52 @@
-{ lib, stdenv, callPackage, fetchFromGitHub
+{ stdenv
+, lib
+, fetchFromGitHub
 , enableStatic ? stdenv.hostPlatform.isStatic
 , enableShared ? !stdenv.hostPlatform.isStatic
+, unstableGitUpdater
+, autoreconfHook
 }:
-let
-  yesno = b: if b then "yes" else "no";
-in stdenv.mkDerivation rec {
+
+stdenv.mkDerivation {
   pname = "libbacktrace";
-  version = "2020-05-13";
+  version = "unstable-2022-12-16";
+
   src = fetchFromGitHub {
     owner = "ianlancetaylor";
-    repo = pname;
-    rev = "9b7f216e867916594d81e8b6118f092ac3fcf704";
-    sha256 = "0qr624v954gnfkmpdlfk66sxz3acyfmv805rybsaggw5gz5sd1nh";
+    repo = "libbacktrace";
+    rev = "da7eff2f37e38136c5a0c8922957b9dfab5483ef";
+    sha256 = "ADp8n1kUf8OysFY/Jv1ytxKjqgz1Nu2VRcFGlt1k/HM=";
   };
+
+  patches = [
+    # Fix tests with shared library.
+    # https://github.com/ianlancetaylor/libbacktrace/pull/99
+    ./0001-libbacktrace-avoid-libtool-wrapping-tests.patch
+
+    # Support multiple debug dirs.
+    # https://github.com/ianlancetaylor/libbacktrace/pull/100
+    ./0002-libbacktrace-Allow-configuring-debug-dir.patch
+    ./0003-libbacktrace-Support-multiple-build-id-directories.patch
+
+    # Support NIX_DEBUG_INFO_DIRS environment variable.
+    ./0004-libbacktrace-Support-NIX_DEBUG_INFO_DIRS-environment.patch
+  ];
```

Comment:
Would not other software need to rely on this for it to be useful. I would not expect other projects to rely on it until it is supported upstream (unless the developer does not realize they are relying on their distro’s patches).

Reply:
In a non-Nix context pkgconfig is often not required because the libraries are in global directories, but within Nix I find that pkgconfig is very helpful to decouple packages' output structures.
I don't expect non-nixpkgs packages to expect pkgconfig for this, but rather it might be helpful for other nixpkgs packages that want to depend on libbacktrace.  I've previously had to generate a pkgconfig manually for it and `overrideAttrs`.

Reply:
> In a non-Nix context pkgconfig is often not required because the libraries are in global directories

Right, and in Nixpkgs, the cc-wrapper plays a similar role.

> I don't expect non-nixpkgs packages to expect pkgconfig for this, but rather it might be helpful for other nixpkgs packages that want to depend on libbacktrace.

We are usually trying to keep packages in Nixpkgs as close to upstream as possible. This typically means limiting patches to build or security fixes and Nix compatibility fixes.

Reply:
Alright, seems sensible.